### DISCUSSION TITLE
Compute zstd compress bound

### DISCUSSION THREAD

**Discussion by tomyl** [klauspost/compress#688] [Ideas/Features]
Hi,

Is it possible to compute a compress bound for zstd? Something like https://pkg.go.dev/github.com/DataDog/zstd?utm_source=godoc#CompressBound

> CompressBound returns the worst case size needed for a destination buffer, which can be used to preallocate a destination buffer or select a previously allocated buffer from a pool.

Thanks,
Tommy

**Comment by klauspost** [comment]
Should be feasible. It will probably be more complex than this with non-default window/block sizes.

You can use a similar function to get a very reasonable buffer size.

Moving to Feature/Idea category.

**Reply by tomyl** [reply]
> You can use a similar function to get a very reasonable buffer size.

Which function is that? Actually I think it's sufficient for me to know how large the destination slice needs to be in order to avoid allocations. ~~I suppose `len(src) + zstd.HeaderMaxSize` should be safe?~~ Hmm no. If I compress 4194304 (4MiB) of random data I get  4194509 bytes back.

**Reply by klauspost** [reply]
I was thinking you could use the `CompressBound`, but it doesn't account for the frame header.

Let me see if I can do something that is closer.

**Reply by klauspost** [reply]
https://github.com/klauspost/compress/pull/691

**Reply by tomyl** [reply]
Nice, thanks! Works well for me.
